targets/TARGET_STM/sleep.c

```diff
@@ -40,17 +40,25 @@ extern void HAL_ResumeTick(void);
 
 void hal_sleep(void)
 {
+    // Disable IRQs
+    core_util_critical_section_enter();
```

Comment:
> So I think 0xc0170 is right - hal_sleep() should probably always be entered with interrupts
> disabled. (And they should have been disabled before checking if we were idle).

Unfortunatley, in this moment `hal_sleep()` does not get called with interrupts already disabled! Not sure if this is a choice or a **bug**, up to you to decide.
Anyway, as it is right now, `mbed-os-5.5` is **not** working correctly non-debug builds, at least for STM-NUCLEO platforms.